be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -470,4 +536,88 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
     else
       Left(error)
   }
+
+  /** checks that for each question, the number of votes is greater or equal than 0, and less that or equal than the number of attendees to the election.
+    *
+    * @param rpcMessage
+    *   the rpc message to validate.
+    * @param result
+    *   the list of the election results.
+    * @param dbActorRef
+    *   the dbActor to ask for the number of attendees.
+    * @param channel
+    *   the channel we need the LAO's data for
+    * @param error
+    *   the error to forward in case where the number of votes is incoherent.
+    * @return
+    *   GraphMessage: passes the rpcMessages to Right if successful else Left with pipeline error
+    */
+  private def checkNumberOfVotes(rpcMessage: JsonRpcRequest, channel: Channel, result: List[ElectionQuestionResult], dbActorRef: AskableActorRef, error: PipelineError): GraphMessage = {
+    val askLaoData = dbActorRef ? DbActor.ReadLaoData(channel)
+    Await.ready(askLaoData, duration).value match {
+      case Some(Success(DbActor.DbActorReadLaoDataAck(laoData: LaoData))) =>
+        val numberOfAttendees = laoData.attendees.size
+        val correct: Boolean =
+          result.forall(electionQuestionResult =>
+            (countNumberOfVotes(electionQuestionResult) <= numberOfAttendees) && allBallotCountsArePositive(electionQuestionResult)
+          )
+        if (correct) {
+          Right(rpcMessage)
+        } else {
+          Left(error)
+        }
+      case Some(Failure(ex: DbActorNAckException)) => Left(PipelineError(ex.code, s"validation of ResultElection failed, could not retrieve laoData : ${ex.message}", rpcMessage.getId))
+      case reply                                   => Left(PipelineError(ErrorCodes.SERVER_ERROR.id, s"ElectionValidator failed : unexpected DbActor reply '$reply'", rpcMessage.getId))
+    }
+  }
+
+  private def countNumberOfVotes(electionQuestionResult: ElectionQuestionResult): Int = {
+    electionQuestionResult.result.map(_.count).sum
+  }
+
+  private def allBallotCountsArePositive(electionQuestionResult: ElectionQuestionResult): Boolean = {
+    electionQuestionResult.result.forall(electionBallotVotes => electionBallotVotes.count >= 0)
+  }
+
+  private def extractElectionQuestions(electionChannel: Channel, dbActorRef: AskableActorRef): Option[List[ElectionQuestion]] = {
```

Comment:
You do not need to define this function. You can use the function `getSetupMessage` (can be found in `ElectionChannel.scala`). See how it is used to extract the questions in `validateCastVoteElection`.